azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/config/AzureComputeServiceContextModule.java

```diff
@@ -99,7 +96,7 @@ protected void configure() {
       });
 
       bind(TemplateOptions.class).to(AzureTemplateOptions.class);
-      bind(PopulateDefaultLoginCredentialsForImageStrategy.class).to(AzurePopulateDefaultLoginCredentialsForImageStrategy.class);
+      bind(TemplateOptionsToStatement.class).to(TemplateOptionsToStatementWithoutPublicKey.class);
```

Comment:
We bind this custom implementation so jclouds does not try to manually edit the `~/.ssh/authorized_keys` if the user provided the public key. Azure already does that.
(Note this class is a copy of the one in the DigitalOcean provider, so once all this stuff is merged I'll move it to jclouds-compute core).